### PR TITLE
HDDS-6543. [Merge rocksdb in datanode] BlockDeletingService adaptation for schema v3 containers.

### PR DESCRIPTION
**Pull request by guihecheng**
## What changes were proposed in this pull request?

BlockDeletingService adaptation for schema v3 containers.

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-6543

## How was this patch tested?

turned on previously ignored UT for schema v3.

### CODE REVIEW

**Comment by guihecheng** [comment]
cc @ChenSammi

File: hadoop-hdds/container-service/src/test/java/org/apache/hadoop/ozone/container/common/TestBlockDeletingService.java [hunk 1]
```diff
@@ -236,6 +241,26 @@ private void createPendingDeleteBlocksSchema2(int numOfBlocksPerContainer,
       int txnID, long containerID, int numOfChunksPerBlock, ChunkBuffer buffer,
       ChunkManager chunkManager, KeyValueContainer container,
       KeyValueContainerData data) {
+    createPendingDeleteBlocksViaTxn(numOfBlocksPerContainer, txnID,
+        containerID, numOfChunksPerBlock, buffer, chunkManager,
+        container, data);
+  }
+
+  @SuppressWarnings("checkstyle:parameternumber")
+  private void createPendingDeleteBlocksSchema3(int numOfBlocksPerContainer,
```
Comment (ChenSammi) [inline hadoop-hdds/container-service/src/test/java/org/apache/hadoop/ozone/container/common/TestBlockDeletingService.java:250]: Looks like createPendingDeleteBlocksSchema3 and createPendingDeleteBlocksSchema2 are identical.

File: hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/keyvalue/statemachine/background/BlockDeletingService.java [hunk 2]
```diff
@@ -516,4 +547,10 @@ public int getPriority() {
       return priority;
     }
   }
+
+  private interface Deleter {
+    void apply(Table<?, DeletedBlocksTransaction> deleteTxnsTable,
+        BatchOperation batch, long txnID)
+        throws IOException;
```
Comment (ChenSammi) [inline hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/keyvalue/statemachine/background/BlockDeletingService.java:554]: L553 and L553 can be merged into one line.

File: hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/metadata/DatanodeStoreSchemaThreeImpl.java [hunk 3]
```diff
@@ -40,7 +40,8 @@
  * - All keys have containerID as prefix.
  * - The table 3 has String as key instead of Long since we want to use prefix.
  */
-public class DatanodeStoreSchemaThreeImpl extends AbstractDatanodeStore {
+public class DatanodeStoreSchemaThreeImpl extends AbstractDatanodeStore
+    implements DeleteTransactionStore<String> {
```
Comment (ChenSammi) [inline hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/metadata/DatanodeStoreSchemaThreeImpl.java:44]: "Rename dropAllWithPrefix  to removeContainerFromDB". Change is missed from last patch.

**Comment by ChenSammi** [comment]
Thanks @guihecheng,  the patch overall looks good.

File: hadoop-hdds/container-service/src/test/java/org/apache/hadoop/ozone/container/common/TestBlockDeletingService.java (hunk 1 above)
Comment (guihecheng) [reply hadoop-hdds/container-service/src/test/java/org/apache/hadoop/ozone/container/common/TestBlockDeletingService.java:250]: Yes, it is, here I just intend to keep the schema check style of function in `createToDeleteBlocks` aligned, but I could remove one of the identical functions and use the lower level function directly when we have schema v2 || v3.

File: hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/keyvalue/statemachine/background/BlockDeletingService.java (hunk 2 above)
Comment (guihecheng) [reply hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/keyvalue/statemachine/background/BlockDeletingService.java:554]: Sure, I'll merge it.

File: hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/metadata/DatanodeStoreSchemaThreeImpl.java (hunk 3 above)
Comment (guihecheng) [reply hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/metadata/DatanodeStoreSchemaThreeImpl.java:44]: Ah, I think I just renamed the helper function in `BlockUtils` but not the one in `DatanodeStoreSchemaThreeImpl`, let's rename it to removeContainerFromDB.

**Comment by ChenSammi** [comment]
+1.

**Pull request merged**